src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -77,7 +77,12 @@ private static BuildEnvironment Initialize()
                 ()=> TryFromFolder(msbuildFromVisualStudioRoot, runningTests, runningInVisualStudio, visualStudioPath),
 
                 // Try from the current directory
-                () => TryFromFolder(currentDirectory, runningTests, runningInVisualStudio, visualStudioPath)
+                () => TryFromFolder(currentDirectory, runningTests, runningInVisualStudio, visualStudioPath),
+
+#if !CLR2COMPATIBILITY // Assemblies compiled against anything older than .NET 4.0 won't have a System.AppContext
+                // Try the base directory that the assembly resolver uses to probe for assemblies.
+                () => TryFromFolder(AppContext.BaseDirectory, runningTests, runningInVisualStudio, visualStudioPath)
```

Comment:
Now that I see what's up, I think this should be above the current-working-directory path. If someone's redistributing an MSBuild runtime, that should be preferred over something that happens to be in the CWD, just as it would be if we found it next to our MSBuild.exe.